components/MoveTimeModal.vue

```diff
@@ -0,0 +1,158 @@
+          <multiselect v-model="transferSplitName" :options="splitOptions" placeholder="Pick a split" class="mb-4"/>
+        </b-col>
+      </b-row>
+      <time-selector v-model="transferTime"/>
+      <div v-b-tooltip.hover :title="invalidTimeTooltip">
+        <b-button @click="chosenSplitFunction" variant="success" class="mt-2 mb-2 text-center" :disabled="!timeValid">
+          Confirm
+        </b-button>
+      </div>
+    </div>
+  </b-modal>
+</template>
+
+<script lang="ts">
+import {
+  secondsToFormattedString,
+  secondsToLivesplitFormat,
+  stringTimeToSeconds
+}                                        from '~/util/durations';
+import {splitFileIsModified, selectTime, Segment, OptionalRealAndGameTime} from '~/util/splits';
+import {Component, Prop, mixins}         from 'nuxt-property-decorator';
+import BaseModal                         from '~/components/BaseModal.vue';
+import Multiselect                       from 'vue-multiselect';
+import {whithLoad}                       from '~/util/loading';
+import store                             from '~/util/store';
+
+@Component({components: {Multiselect}})
+export default class MoveTimeModal extends mixins(BaseModal) {
+  modalRef: string = 'MoveTimeModal';
+
+  transferTime: number = 0;
+
+  transferSplitName: string = "";
+
+  @Prop()
+  splits!: Segment[];
+
+  @Prop()
+  currentSplitIndex!: number;
+
+  get segmentGold() {
+    const gold = selectTime(this.currentSplit.BestSegmentTime);
+    return stringTimeToSeconds(gold || "0.0.0.0");
+  }
+
+  get splitOptions() {
+    if (this.currentSplitIndex == 0)
+      return [this.nextSplit.Name];
+
+    const lastSplitIndex = this.splits.length - 1;
+    if (this.currentSplitIndex == lastSplitIndex)
+      return [this.previousSplit.Name];
+
+    return [this.previousSplit.Name, this.nextSplit.Name];
+  }
+
+  get timeValid() {
+    if (this.transferTime == 0 || this.transferTime > this.segmentGold || this.transferSplitName == '') {
+      return false;
+    }
+
+    return true;
+  }
+
+  get invalidTimeTooltip() {
+    if (this.transferTime == 0)
+      return 'You must add a time.';
+
+    if (this.transferSplitName == '')
+      return 'You need select a split to transfer to.';
+
+    if (this.transferTime > this.segmentGold)
+      return `You cannot transfer more than ${secondsToFormattedString(this.segmentGold)}.`
+
+    return "";
+  }
+
+  get currentSplit() {
+    return this.splits[this.currentSplitIndex];
+  }
+
+  get nextSplit() {
+    return this.splits[this.currentSplitIndex + 1];
+  }
+
+  get previousSplit() {
+    return this.splits[this.currentSplitIndex - 1];
+  }
+
+  get chosenSplitFunction() {
+    if (this.transferSplitName === this.nextSplit?.Name)
+      return () => this.moveTimeNext();
+    
+    if (this.transferSplitName === this.previousSplit?.Name)
+      return () => this.moveTimePrevious();
+  }
+
+  moveTransferTime(chosenTime: OptionalRealAndGameTime, isCurrentSplit: boolean) {
```

Comment:
The trick of making `transferTime` negative is clever, but the resulting code is not very easy to use (and will be hard to move to a web worker since you're using `this.transferTime` which won't exist in the scope of the worker because `this` won't make sense. I suggest passing `transferTime` as a parameter instead of `isCurrentSplit`, and therefore in `moveTime{Previous/Next}` you can do 
```TS
this.moveTransferTime(this.currentSplit.BestSegmentTime, -this.transferTime);
``` 
for example